frontend/packages/ceph-storage-plugin/src/components/attached-devices-mode/lso-disk-inventory/state-reducer.ts

```diff
@@ -0,0 +1,83 @@
+export enum ActionType {
+  SET_ALERTS_MAP = 'SET_ALERTS_MAP',
+  SET_TEMPLATES_MAP = 'SET_TEMPLATES_MAP',
+  SET_METRICS_MAP = 'SET_METRICS_MAP',
+  SET_IS_REBALANCING = 'SET_IS_REBALANCING',
+  SET_REPLACING_DISK = 'SET_REPLACING_DISK',
+}
+
+export const initialState: OCSColumnState = {
+  metricsMap: {},
+  templateMap: {},
+  alertsMap: {},
+  isRebalancing: false,
+  replacingDisk: { disk: '', osd: '' },
+};
+
+export const reducer = (state: OCSColumnState, action: OCSColumnStateAction) => {
+  switch (action.type) {
+    case ActionType.SET_ALERTS_MAP: {
+      return {
+        ...state,
+        alertsMap: action.payload,
+      };
+    }
+    case ActionType.SET_METRICS_MAP: {
+      return {
+        ...state,
+        metricsMap: action.payload,
+      };
+    }
+    case ActionType.SET_REPLACING_DISK: {
+      return {
+        ...state,
+        replacingDisk: action.payload,
+      };
+    }
+    case ActionType.SET_IS_REBALANCING: {
+      return {
+        ...state,
+        isRebalancing: action.payload,
+      };
+    }
+    default:
+      return state;
+  }
+};
+
+export type OCSColumnStateAction =
+  | { type: ActionType.SET_ALERTS_MAP; payload: OCSDiskList }
+  | { type: ActionType.SET_TEMPLATES_MAP; payload: OCSDiskList }
+  | { type: ActionType.SET_METRICS_MAP; payload: OCSDiskList }
+  | { type: ActionType.SET_REPLACING_DISK; payload: { disk: string; osd: string } }
+  | { type: ActionType.SET_IS_REBALANCING; payload: boolean };
+
+export type OCSColumnState = {
+  metricsMap: OCSDiskList;
+  templateMap: OCSDiskList;
+  alertsMap: OCSDiskList;
+  isRebalancing: boolean;
+  replacingDisk: {
+    disk: string;
+    osd: string;
+  };
+};
+
+export type OCSDiskList = {
+  [diskName: string]: OCSDiskMetadata;
+};
+
+export type OCSDiskMetadata = { osd: string; status: OCSDiskStatus };
+
+export type OCSDiskStatus =
+  | 'Online'
+  | 'Offline'
+  | 'Not responding'
+  | 'Preparing to replace'
+  | 'Replacement ready'
+  | 'Replacement failed'
+  | 'Unknown';
+
+export type AlertsDiskMap = {
+  [x: string]: string;
```

Comment:
```suggestion
  [key: string]: string;
```
Or something meaningful `x` doesn't tell me anything.